prov/efa/src/efa_mr.c

```diff
@@ -397,14 +397,21 @@ static int efa_mr_dereg_impl(struct efa_mr *efa_mr)
 	struct efa_domain *efa_domain;
 	int ret = 0;
 	int err;
+	size_t ibv_mr_size;
 
 	efa_domain = efa_mr->domain;
 	if (efa_mr->ibv_mr) {
+		ibv_mr_size = efa_mr->ibv_mr->length;
 		err = -ibv_dereg_mr(efa_mr->ibv_mr);
 		if (err) {
 			EFA_WARN(FI_LOG_MR,
 				"Unable to deregister memory registration\n");
 			ret = err;
+		} else {
+			efa_mr->domain->ibv_mr_reg_ct--;
+			efa_mr->domain->ibv_mr_reg_sz -= ibv_mr_size;
+			EFA_INFO(FI_LOG_MR, "Deregistered memory of size %lu for ibv pd %p, total mr reg size %lu, mr reg count %lu\n",
```

Comment:
nit: `%zu` for `size_t`
```suggestion
			EFA_INFO(FI_LOG_MR, "Deregistered memory of size %zu for ibv pd %p, total mr reg size %zu, mr reg count %zu\n",
```

Reply:
Updated.